app/components/SingleProduct.jsx

```diff
@@ -1,17 +1,118 @@
 import React from 'react'
-
 import { login } from 'APP/app/reducers/auth'
 import { connect } from 'react-redux'
 
-const SingleProduct = (props) => (
-  <div>
-    <h2>Product name</h2>
-    <h3>Rest of the product description</h3>
-  </div>
+// *------------------- COMPONENT -----------------*//
+
+class SingleProduct extends React.Component {
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      color: '',
+      size: '',
+      quantity: 0
+    }
+    this.setColor = this.setColor.bide(this)
+    this.setSize = this.setSize.bide(this)
+    this.setQuantity = this.setQuantity.bide(this)
+  }
+
+  setColor(val){
+    this.setState({ color: val })
+  }
+
+  setSize(val){
+    this.setState({ size: val })
+  }
+
+  setQuantity(val){
+    this.setState({ quantity: val})
+  }
+
+  render() {
+    const product = this.props.selectedProduct
+    const reviews = this.props.reviews
+
+    return (
+      <div>
+        <div className = "col-md-6">
+          <div className="carousel carousel-slider">
+            <a className="carousel-item" href="#one!">
+              {
+                product && product.images.map(image => (
+                  <img src={ image } />
+                ))
+              }
+            </a>
+          </div>
+        </div>
+
+        <div className="col-md-6">
+          <form onSubmit = { this.props.handleSubmit }>
```

Comment:
"this.props.handleSubmit" should be "this.props.onSubmitHandle" because of how you wrote it in the connect dispatch.